packages/storage/amplify_storage_s3/example/integration_test/main_test.dart

```diff
@@ -0,0 +1,623 @@
+// Copyright 2022 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import 'dart:async';
+import 'dart:convert';
+
+import 'package:amplify_auth_cognito/amplify_auth_cognito.dart';
+import 'package:amplify_core/amplify_core.dart';
+import 'package:amplify_secure_storage/amplify_secure_storage.dart';
+import 'package:amplify_storage_s3/amplify_storage_s3.dart';
+import 'package:amplify_storage_s3_example/amplifyconfiguration.dart';
+import 'package:drift/drift.dart';
+import 'package:flutter_test/flutter_test.dart';
+import 'package:http/http.dart' as http;
+import 'package:integration_test/integration_test.dart';
+
+import 'download_file_tester/download_file_tester.dart';
+
+class CustomPrefixResolver implements S3PrefixResolver {
+  const CustomPrefixResolver();
+
+  @override
+  Future<String> resolvePrefix({
+    required StorageAccessLevel accessLevel,
+    String? identityId,
+  }) async {
+    final currentUserIdentityId =
+        ((await Amplify.Auth.fetchAuthSession()) as CognitoAuthSession)
+            .identityId;
+    switch (accessLevel) {
+      case StorageAccessLevel.guest:
+        return 'everyone/';
+      case StorageAccessLevel.protected:
+        return 'shared/${identityId ?? currentUserIdentityId}/';
+      case StorageAccessLevel.private:
+        return 'private/$currentUserIdentityId/';
+    }
+  }
+}
+
+void main() async {
```

Comment:
This file is pretty large. Can we break up the tests into different files by some sort of grouping? In auth, it's by API although I understand if some other scheme makes sense here.

Reply:
The test cases depend on each other, e.g. upload an object then do some operations on it, so I kept it in the same file. And it's not really that large 😅 can navigate between tests easily via IDE file outline as well.